### PR TITLE
EC HA Support

### PR DESCRIPTION
**Pull request by patrick-wright**

### CODE REVIEW

File: libraries/chef_package_helper.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ def install_product(resource)
     #
     #
     def reconfigure_product(resource)
+      install_mixlib_versioning
       chef_ingredient resource.product_name do
         action :reconfigure
```
Comment (jtimberman) [inline libraries/chef_package_helper.rb]: This shouldn't [be necessary](https://github.com/chef-cookbooks/chef-ingredient/blob/master/libraries/chef_ingredient_provider.rb#L65-L66), though?

Comment (patrick-wright) [inline libraries/chef_package_helper.rb]: My thinking was that recipes would typically install a package using chef_ingredient before attempting to reconfigure.  However, I am using `package` to install a `remote_file` and I don't want to deal with the `product_lookup` logic.  Later I reconfigure that product and I was having to call `install_mixlib_versioning()`in my wrapper cookbook.  I would like to see this stay.

Comment (patrick-wright) [inline libraries/chef_package_helper.rb]: oh duh, this is in my cookbook.  I can remove this now.

File: recipes/ha-enterprise-chef-backend.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  only_if { ::File.exists?(::File.join(current_server.config_path, 'pivotal.pem')) }
+end
+
+package 'drbd8-utils'
+
+package 'gcc'
+
+package 'flex'
+
+package 'make'
```
Comment (jtimberman) [inline recipes/ha-enterprise-chef-backend.rb]: I think you mean:

```ruby
include_recipe 'build-essential'
```

:cake:

File: recipes/ha-enterprise-chef-backend.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      raise exception unless ctl.stdout.include?('Press CTRL-C to abort')
+    end
+  end
+  not_if { ::File.exist?('/var/opt/opscode/drbd/drbd_ready') }
+end
+
+# Ubuntu 12.04 step not included in docs
+# "on" nodes must match `hostname` value, so we strip the domain
+execute 'remove domain from pc0.res' do
+  command 'sed -i s/.us-west-2.compute.internal/\/ /var/opt/opscode/drbd/etc/pc0.res'
```
Comment (jtimberman) [inline recipes/ha-enterprise-chef-backend.rb]: I think `node['hostname']` won't have the domain by default on EC2 nodes?

File: recipes/ha-enterprise-chef-backend.rb [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  only_if 'cat /var/opt/opscode/drbd/etc/pc0.res | grep ".us-west-2.compute.internal"'
```
Comment (jtimberman) [inline recipes/ha-enterprise-chef-backend.rb]:
```
only_if 'grep ".us-west-2.compute.internal" /var/opt/opscode/drbd/etc/pc0.res'
```

;)

File: recipes/ha-enterprise-chef-backend.rb (hunk 2 above)
Comment (patrick-wright) [inline recipes/ha-enterprise-chef-backend.rb]: thank you.  I actually already install build-essential so I'll try removing these package installs.

File: recipes/ha-enterprise-chef-backend.rb (hunk 3 above)
Comment (patrick-wright) [inline recipes/ha-enterprise-chef-backend.rb]: This is the root issue: https://github.com/chef/chef-server/blob/fce335d0f84cf042ba291e2cfe7aa9204e1188d6/omnibus/files/private-chef-cookbooks/chef-ha-drbd/templates/default/pc0.res.erb#L6

File: recipes/ha-enterprise-chef-backend.rb (hunk 4 above)
Comment (patrick-wright) [inline recipes/ha-enterprise-chef-backend.rb]: ty

**Pull request merged**